src/DcaRelationsManager.php

```diff
@@ -351,9 +353,52 @@ public function getRelatedRecords(mixed $value, DataContainer $dc): mixed
         return $value;
     }
 
-    #[AsHook('sqlGetFromFile')]
+    public function appendToSchema(Schema $schema): void
+    {
+        foreach ($this->connection->createSchemaManager()->listTables() as $table) {
+            $tableName = $table->getName();
+
+            if (!str_starts_with($tableName, 'tl_')) {
+                continue;
+            }
+
+            Controller::loadDataContainer($tableName);
+
+            if (!isset($GLOBALS['TL_DCA'][$tableName]['fields'])) {
+                continue;
+            }
+
+            foreach (array_keys($GLOBALS['TL_DCA'][$tableName]['fields']) as $fieldName) {
+                $relation = $this->getRelation($tableName, $fieldName);
+
+                if (null === $relation || $relation['skipInstall']) {
+                    continue;
+                }
+
+                $referenceType = $relation['reference_definition']['type'];
+                unset($relation['reference_definition']['type']);
+
+                $relatedType = $relation['related_definition']['type'];
+                unset($relation['related_definition']['type']);
+
+                $schemaTable = $schema->hasTable($relation['table']) ? $schema->getTable($relation['table']) : $schema->createTable($relation['table']);
+                $schemaTable->addColumn($relation['reference_field'], $referenceType, $relation['reference_definition']);
+                $schemaTable->addColumn($relation['related_field'], $relatedType, $relation['related_definition']);
+
+                $indexName = $relation['reference_field'].'_'.$relation['related_field'];
+
+                // Add the index only if there is no other (avoid duplicate keys)
+                if (!$schemaTable->hasIndex($indexName)) {
+                    $schemaTable->addUniqueIndex([$relation['reference_field'], $relation['related_field']], $indexName);
+                }
+            }
+        }
+    }
+
     public function addRelationTables(array $definitions): array
```

Comment:
Honestly, this was a hook listener, no? Never intended for public API. Imho you can break with this and release this whole thing as a bugfix to 5.0 because it was meant to be removed from the beginning and the fact that the relations do not work is a bug.